lib/technical.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const request = require('request-promise-native');
+
+module.exports = config => {
+  const util = require('./util')(config);
+
+  /**
+   * Util function to get the sector performance data.
+   * 
+   * @returns {Promise}
+   *   The request promise.
+   */
+  const sma = (symbol, interval, time_period, series_type) =>
```

Comment:
I do like the pattern you have in lib/data.js with the `timeSeries` function, and then have that called with different `fn` calls below.  Are you thinking eventually that would happen with the technical indicators?

Maybe `const sma` should be `const technicalIndicator`  which is then used by the various api methods?

Reply:
I see what you were saying in your previous comment that each endpoint would provide the specific parameters needed.   That makes sense, I was just hoping for less typing :)

I think this looks good.   Once you merge in let's discuss how to break up the rest of the endpoints?

Reply:
Yeah, I think the `lib/technical.js` file will contain a bunch of generic functions, similar to `timeSeries` from `lib/data.js`, but aimed at the technical indicators. I've implemented the first one, SMA, which can also be used for EMA, WMA, DEMA, TEMA, TRIMA and KAMA. I named it SMA, because thats the first function signature it was meant to match, I didnt really want to name it something like `technicalFn1`.

If we follow the same pattern (using the list given on the documentation page), the next function will be mama with the following signature:

```js
const mama = (symbol, interval, series_type, fastlimit, slowlimit) => {}
```

then T3 could also use the SMA function, and MACD would have a new signature; so on and so fourth until the end of the list..

Thoughts?

Reply:
This method makes the implementation a bit weird, due to how the urls are being build with dynamic parameters, but it should yield a pretty lean sdk which is better imo.

Reply:
Ya... I think it's one of those things that feels funky (given the api params) but will probably be just fine in implementation.